### PR TITLE
Allow startup without working Elasticsearch cluster

### LINKED ISSUE #1136: Graylog startup behaviour
Currently when a server node starts and finds issues with the Elasticsearch cluster it refuses to complete the startup process - with the introduction of the journaling subsystem in 1.0 I find this behaviour a bit problematic since it could mean loosing messages.

I would like to suggest that server nodes **always** start regardless of the ES cluster status (and thus can buffer messages in it's journal) - the web interface could then inform the user that ES has a problem that needs investigating to avoid confusion.

### PR DESCRIPTION
**Pull request by joschi**
The previous implementation of `IndexerSetupService` (and related classes) would shutdown Graylog if Elasticsearch wasn't available or the cluster health state was RED on startup. This made sense as long as Graylog didn't have a proper on-disk journal in which messages could be persistently buffered.

The new implementation simply acknowledges the unavailability of Elasticsearch and generates an appropriate system notification (and log entries) but does not shutdown Graylog.

Fixes #1136

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/initializers/IndexerSetupService.java [hunk 1]
```diff
@@ -114,11 +111,10 @@ protected void startUp() throws Exception {
         Tools.silenceUncaughtExceptionsInThisThread();
 
         LOG.debug("Starting indexer");
-        try {
-            node.start();
+        node.start();
```
Comment (kroepke) [inline graylog2-server/src/main/java/org/graylog2/initializers/IndexerSetupService.java]: are we sure this and the next line never throw exceptions?
if they do, we would miss them and not print any error courtesy of `Tools.silenceUncaughtExceptionsInThisThread();`

**Comment by kroepke** [comment]
other than the comment above this looks really good :+1:

File: graylog2-server/src/main/java/org/graylog2/initializers/IndexerSetupService.java (hunk 1 above)
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/initializers/IndexerSetupService.java]: The behavior is the same as before as the outer most `catch` block just re-threw the exception: https://github.com/Graylog2/graylog2-server/pull/1289/files#diff-72fd860130aebd46f43cb3fa1fb32d18L193

File: graylog2-rest-client/src/main/java/org/graylog2/restclient/models/Notification.java [hunk 2]
```diff
@@ -28,6 +28,8 @@
         MULTI_MASTER,
         NO_MASTER,
         ES_OPEN_FILES,
+        ES_CLUSTER_RED,
+        ES_UNAVAILABLE,
         NO_INPUT_RUNNING,
```
Comment (bernd) [inline graylog2-rest-client/src/main/java/org/graylog2/restclient/models/Notification.java]: This requires an update to the web interface as well, right?

Comment (joschi) [inline graylog2-rest-client/src/main/java/org/graylog2/restclient/models/Notification.java]: Yes, see Graylog2/graylog2-web-interface#1516.

**Comment by bernd** [comment]
I tested the branch with Elasticsearch 1.5. Seems to work fine!

I am getting the following message in the logs on a regular basis. Maybe we should have an "es alive" check in the `IndexRotationThread` as well? It curently just checks if the `IndexerSetupService` is running.

```
2015-07-07 11:18:17,557 ERROR: org.graylog2.periodical.IndexRotationThread - Couldn't point deflector to a new index
org.elasticsearch.discovery.MasterNotDiscoveredException: waited for [30s]
	at org.elasticsearch.action.support.master.TransportMasterNodeOperationAction$4.onTimeout(TransportMasterNodeOperationAction.java:164)
	at org.elasticsearch.cluster.ClusterStateObserver$ObserverClusterStateListener.onTimeout(ClusterStateObserver.java:231)
	at org.elasticsearch.cluster.service.InternalClusterService$NotifyTimeout.run(InternalClusterService.java:560)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by joschi** [comment]
@bernd Yes, thought about that as well. I would add this in another PR, though.

**Comment by bernd** [comment]
> Yes, thought about that as well. I would add this in another PR, though. 

Ack! LGTM :+1:

**Pull request merged**